server/list.go

```diff
@@ -235,34 +235,34 @@ func (l *listManager) ChangeAssignment(issueID string, userID string, sendTo str
 
 	if userID == sendTo && list == OutListKey {
 		if err := l.store.RemoveReference(userID, issueID, OutListKey); err != nil {
-			return "", "", err
+			return nil, "", err
 		}
 
 		if err := l.store.AddReference(userID, issueID, MyListKey, "", ""); err != nil {
-			return "", "", err
+			return nil, "", err
 		}
 
-		return issue.Message, ir.ForeignUserID, nil
+		return issue, ir.ForeignUserID, nil
 	}
 
 	if err := l.store.RemoveReference(userID, issueID, list); err != nil {
-		return "", "", err
+		return nil, "", err
 	}
 
-	receiverIssue := newIssue(issue.Message, issue.Description, issue.PostID)
+	receiverIssue := newIssue(issue.Message, issue.PostPermalink, issue.Description, issue.PostID)
```

Comment:
Seems like this could just accept a `issue` instead of these 4 fields derived from the `issue`

Reply:
@mickmister This function https://github.com/mattermost/mattermost-plugin-todo/blob/b8dff94f7bebfcb7f81f36f588f110e08e34ef9f/server/issue.go#L27 just return a new object with passed parameters value. At some places https://github.com/mattermost/mattermost-plugin-todo/blob/b8dff94f7bebfcb7f81f36f588f110e08e34ef9f/server/list.go#L62 we are passing the value directly instead of the case above